Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -183,7 +184,8 @@ class BrowserViewController: UIViewController {
             home.view.setNeedsUpdateConstraints()
         }
 
-        if let tab = tabManager.selectedTab,
+        if !isInBackground,
```

Comment:
This seems arbitrary putting this here -- what is it supposed to be protecting? From your comments, it sounds like we might be crashing at `profile.bookmarks.modelFactory` in `updateURLBarDisplayURL`. Would it make more sense to put the guard there since there are several other places we call `updateURLBarDisplayURL`?

But even then, we access the DB in many places throughout the app, so it's unclear to me what's special about this particular case.